src/Codec/Xlsx/Types/DataValidation.hs

```diff
@@ -0,0 +1,200 @@
+    { _dvAllowBlank       :: Maybe Bool
+    , _dvError            :: Maybe Text
+    , _dvErrorStyle       :: Maybe ErrorStyle
+    , _dvErrorTitle       :: Maybe Text
+    , _dvPrompt           :: Maybe Text
+    , _dvPromptTitle      :: Maybe Text
+    , _dvShowDropDown     :: Maybe Bool
+    , _dvShowErrorMessage :: Maybe Bool
+    , _dvShowInputMessage :: Maybe Bool
+    , _dvSqref            :: SqRef
+    , _dvValidationType   :: Maybe ValidationType
+    } deriving (Eq, Show)
+
+makeLenses ''DataValidation
+
+instance Default DataValidation where
+    def = DataValidation
+      Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing
+      (SqRef []) Nothing
+
+{-------------------------------------------------------------------------------
+  Parsing
+-------------------------------------------------------------------------------}
+
+instance FromAttrVal ErrorStyle where
+    fromAttrVal "information" = readSuccess ErrorStyleInformation
+    fromAttrVal "stop"        = readSuccess ErrorStyleStop
+    fromAttrVal "warning"     = readSuccess ErrorStyleWarning
+    fromAttrVal t             = invalidText "ErrorStyle" t
+
+instance FromCursor DataValidation where
+    fromCursor cur = cur $/ element (n_ "dataValidation") >=> fromCur
+      where
+        fromCur c = do
+            _dvAllowBlank       <- maybeAttribute "allowBlank"       c
+            _dvError            <- maybeAttribute "error"            c
+            _dvErrorStyle       <- maybeAttribute "errorStyle"       c
+            _dvErrorTitle       <- maybeAttribute "errorTitle"       c
+            mop                 <- maybeAttribute "operator"         c
+            _dvPrompt           <- maybeAttribute "prompt"           c
+            _dvPromptTitle      <- maybeAttribute "promptTitle"      c
+            _dvShowDropDown     <- maybeAttribute "showDropDown"     c
+            _dvShowErrorMessage <- maybeAttribute "showErrorMessage" c
+            _dvShowInputMessage <- maybeAttribute "showInputMessage" c
+            _dvSqref            <- fromAttribute  "sqref"            c
+            mtype               <- maybeAttribute "type"             c
+            _dvValidationType   <- readValidation mop mtype          c
+            return DataValidation{..}
+
+readValidation :: Maybe Text -> Maybe Text -> Cursor -> [Maybe ValidationType]
+readValidation _ Nothing         _   = return $ Nothing
+readValidation _ (Just "none")   _   = return $ Just ValidationTypeNone
+readValidation _ (Just "custom") cur = do
+    f <- fromCursor cur
+    return $ Just $ ValidationTypeCustom f
+readValidation _ (Just "list") cur = do
+    f  <- cur $/ element (n_ "formula1") >=> fromCursor
+    as <- readListFormula f
+    return $ Just $ ValidationTypeList as
+readValidation (Just op) (Just ty) cur = do
+    opExp <- readOpExpression2 op cur
+    Just <$> readValidationTypeOpExp ty opExp
+readValidation _ _ _ = []
+
+readListFormula :: Formula -> [[Text]]
```

Comment:
For `Text` there is https://www.stackage.org/haddock/lts-7.5/xml-conduit-1.3.5/Text-XML-Cursor.html#v:content but probably this point isn't that important